adaptivefiltering/segmentation.py

```diff
@@ -60,3 +65,199 @@ def __geo_interface__(self):
             "type": "FeatureCollection",
             "features": self.features,
         }
+
+
+class Interactive_Map:
+    def __init__(self, dataset):
+        """This class manages the interactive map on which one can choose the segmentation.
+
+        :param dataset:
+            The dataset from which the map should be displayed. This needs to be in a valid georeferenced format. Eg.: EPSG:4326
+        :type dataset: Dataset
+
+        """
+        self.dataset = dataset
+        self.coordinates_mean, self.polygon_boundary = self.get_bounds()
+
+        self.m = ipyleaflet.Map(
+            basemap=ipyleaflet.basemaps.Esri.WorldImagery,
+            center=(self.coordinates_mean[0], self.coordinates_mean[1]),
+            crs=ipyleaflet.projections.EPSG3857,
+            scroll_wheel_zoom=True,
+            max_zoom=20,
+        )
+        self.m.add_layer(self.polygon_boundary)
+
+        # always add polygon draw tool and zoom slider
+        self.add_zoom_slider()
+        self.add_polygon_control()
+
+        # get a color picker to put on the right side of the map
+        self.color_pick = self.get_color_pick()
+        # setup the grid with a list of widgets
+        self.setup_grid([self.m, self.color_pick])
+
+    def get_bounds(self):
+        """takes the boundry coordinates of given dataset through the hexbin filter and returns them as a polygon
+
+        :return:
+        :param coordinates_mean:
+            the approximate center of the polygon, this is where the map will be centered
+        :type coordinates_mena: ndarray
+
+
+        :param polygon_boundary:
+            An ipyleaflet Polygon object which can be added to the map to mark the selected area.
+        :type polygon_boundary: ipyleaflet Polygon
+
+        """
+        hexbin_json = [
+            self.dataset.filename,
+            {"type": "filters.reprojection", "out_srs": "EPSG:4326"},
+            {"type": "filters.hexbin"},
+        ]
+        hexbin_pipeline = pdal.Pipeline(json.dumps(hexbin_json))
+        hexbin_pipeline.execute()
+        hexbin_geojson = json.loads(hexbin_pipeline.metadata)["metadata"][
+            "filters.hexbin"
+        ]["boundary_json"]
+        # get the previous coordinate representation
+        boundary_coordinates = hexbin_geojson["coordinates"][0]
+        coordinates_mean = np.mean(boundary_coordinates, axis=0)
+        polygon_boundary = ipyleaflet.Polygon(
+            locations=boundary_coordinates, color="gray", opacity=0.9
+        )
+        return coordinates_mean, polygon_boundary
+
+    def add_zoom_slider(self):
```

Comment:
I am not sure it is worth to have these methods if we always want to call them exactly once. Comments?

Reply:
I was once taught that the init of a class should be as compact as feasible, so i tried to wrap most things in a function to not clutter the init and also provide a way for a bit more documentation for each of these functions. 
One can now also easily minimize them to make the class a bit less crowed. 

that was my reasoning behind the functions.